### PR TITLE
Documentation for babel-plugin-transform-runtime

### PR DESCRIPTION
**Pull request by trodrigues**
Based on https://github.com/babel/babel.github.io/blob/862b43db93e48762671267034a50c30c00e433e2/docs/usage/runtime.md

Not sure if the documentation from Babel 5 is still 100% accurate, but from what I could understand it seems to be.

Related to https://github.com/babel/babel.github.io/issues/674 and https://github.com/babel/babel.github.io/issues/491

Also added it to https://github.com/babel/babel.github.io/pull/678

### CODE REVIEW

File: packages/babel-plugin-transform-runtime/README.md [hunk 1]
```diff
@@ -33,3 +45,135 @@ require("babel-core").transform("code", {
...
+"use strict";
+
+var _core = require("babel-runtime/core-js");
+
+var sym = _core.Symbol();
+
+var promise = new _core.Promise();
+
+console.log(_core.$for.getIterator(arr));
+```
```
Comment (zloirock) [inline packages/babel-plugin-transform-runtime/README.md]: Have you tested it? :) These examples are obsolete for a long time.

Comment (trodrigues) [inline packages/babel-plugin-transform-runtime/README.md]: Like I said on my description (later edit) I mostly copied this from the old docs. I would also be ok with removing the technical details part and leave that to someone who might actually know what the details are and just leave the initial description, as right now there's nothing, and when I was trying to use this plugin it took me quite a bit of digging to understand what it was :(

Comment (zloirock) [inline packages/babel-plugin-transform-runtime/README.md]: You can just add actual result for these examples.

Comment (trodrigues) [inline packages/babel-plugin-transform-runtime/README.md]: Ahh good point :sweat_smile: I'll do that

**Comment by trodrigues** [comment]
Updated code examples according to @zloirock feedback

**Comment by geminiyellow** [comment]
:+1:

File: packages/babel-plugin-transform-runtime/README.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
-Externalise references to helpers and builtins, automatically polyfilling your code without polluting globals
+Externalise references to helpers and builtins, automatically polyfilling your code without polluting globals.
+
+## Why?
+
+Babel uses very small helpers for common functions such as `_extend`. By default this will be added to every file that requires it. This duplication is sometimes unnecessary, especially when your application is spread out over multiple files.
+
+This is where the runtime transformer plugin comes in: all of the helpers will reference the module babel-runtime to avoid duplication across your compiled output. The runtime will be compiled into your build.
+
+Another purpose of this transformer is to create a sandboxed environment for your code. Built-ins such as `Promise`, `Set` and `Map` are aliased to `core-js` so you can use them seamlessly without having to require a globally polluting [polyfill](https://babeljs.io/docs/usage/polyfill/).
```
Comment (ForbesLindesay) [inline packages/babel-plugin-transform-runtime/README.md]: Is it worth explicitly stating that babel won't add these polyfills by default?

Comment (trodrigues) [inline packages/babel-plugin-transform-runtime/README.md]: That's a good idea, I've seen people confused about this.

**Comment by trodrigues** [comment]
Added an additional tidbit to make it clear `babel-runtime` needs to be a dependency of the project.

**Comment by winglian** [comment]
Currently `babel-plugin-transform-runtime` requires `babel-runtime@^5.0.0`, but simply running `npm i babel-runtime --save` installs the latest `6.3.19`. Should we be installing with `npm i babel-runtime@5 --save` or should the dependency be updated?

**Comment by loganfsmyth** [comment]
@winglian Neither, the dependency is there because the transform depends on the 5.x version. Your code should depend on the 6.x version.

**Comment by trodrigues** [comment]
I've rebased this branch on master and squashed the commits made so far.

Any further feedback or anything else I could do to have this merged?

File: packages/babel-plugin-transform-runtime/README.md [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+Externalise references to helpers and builtins, automatically polyfilling your code without polluting globals.
+
+## Why?
+
+Babel uses very small helpers for common functions such as `_extend`. By default this will be added to every file that requires it. This duplication is sometimes unnecessary, especially when your application is spread out over multiple files.
+
+This is where the runtime transformer plugin comes in: all of the helpers will reference the module babel-runtime to avoid duplication across your compiled output. The runtime will be compiled into your build.
+
+Another purpose of this transformer is to create a sandboxed environment for your code. If you use [babel-polyfill](https://github.com/contentful/contentful-space-sync/releases/tag/v3.0.0) and the built-ins it provides such as `Promise`, `Set` and `Map`, those will pollute the global scope. While this might be ok for an app or a command line tool, it becomes a problem if your code is a library which you intend to publish for others to use or if you can't exactly control the environment in which your code will run.
```
Comment (amasad) [inline packages/babel-plugin-transform-runtime/README.md]: The link looks incorrect here

Comment (trodrigues) [inline packages/babel-plugin-transform-runtime/README.md]: Oh boy, that's a complete copy paste failure, thanks for noticing 😅

Comment (trodrigues) [inline packages/babel-plugin-transform-runtime/README.md]: Also fixed it on the other PR to the website.

**Pull request merged**